src/api/server.middleware.ts

```diff
@@ -83,6 +84,22 @@ export class ServerMiddleware {
     return result;
   }
 
+  public static mapToServerModel(schedule: ScheduleDataModel): ServerDataModel {
+    const month_number = schedule.schedule_info.month_number;
+    const publicHolidaysLogic = new PublicHolidaysLogic(schedule.schedule_info.year.toString());
+    const result = _.cloneDeep(schedule) as ServerDataModel;
+
+    const holidays = schedule.month_info.dates.filter((d) =>
+      publicHolidaysLogic.isPublicHoliday(d, month_number)
+    );
+    const indexedDays = schedule.month_info.dates.map((d, index) => [d, index]);
+    const indexedHolidays = holidays
+      .map((h) => indexedDays.filter((d) => d[0] === h).map((d) => d[1]))
+      .map((h) => h[0]);
+    result.month_info.holidays = indexedHolidays;
```

Comment:
próbowałam dopasować indeksy zgodnie z [tym](https://github.com/Project-Summer-AI-Lab-Glider/nurse-scheduling-problem-frontend/pull/349#discussion_r624014861), ale nie jestem pewna, czy to o to chodziło

Reply:
O to chodziło, ale trochę źle to robisz:
ten warunek
`indexedDays.filter((d) => d[0] === h)`
zwórci dwa dni jeśli mamy tydzień który zachodzi z następnego miesiąca i święta w pierwszym tygodniu. 
(jak np. w maju)

Żeby tego uniknąć potrzeb stworzyć `MonthLogic` i wyciągnąć z niej `verboseDates`. 

Plus mi się wydaje, że trochę komplikujesz z mapowaniem i filtrowaniem i można to zrobić prościej:
```ts
const holidays = verboseDates.map((d,i) => [d,i])
                                                   .filter(dayIndPair =>  dayIndPair[0].isPublicHolday)
                                                   .map(dayIndPair => dayIndPair[1])
```